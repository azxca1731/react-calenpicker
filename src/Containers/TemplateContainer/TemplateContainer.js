import React from "react";
import PropTypes from "prop-types";

import { CssConnector, PropsConnector, DayConnector } from "Containers/Provider";
import Template from "Components/Template";

class TemplateContainer extends React.Component {
  constructor(props) {
    super(props);
    this.element = React.createRef();
  }

  componentDidMount() {
    if (this.props.canMouseWheel) {
      this.element.current.addEventListener("wheel", this.handleMouseEvent);
    }
  }

  componentWillUnmount() {
    if (this.props.canMouseWheel) {
      this.element.current.removeEventListener("wheel", this.handleMouseEvent);
    }
  }

  handleMouseEvent = event => {
    if (event && event.deltaY > 0) {
      this.props.increaseMonth();
    } else if (event && event.deltaY < 0) {
      this.props.decreaseMonth();
    }
  };

  render() {
    return (
      <div ref={this.element}>
        <Template {...this.props} />
      </div>
    );
  }
}

TemplateContainer.propTypes = {
  cssObject: PropTypes.object,
  decreaseMonth: PropTypes.func,
  increaseMonth: PropTypes.func,
  canMouseWheel: PropTypes.bool
};

export default PropsConnector(({ state }) => ({
  canMouseWheel: state.canMouseWheel
}))(
  DayConnector(({ actions }) => ({
    decreaseMonth: actions.decreaseMonth,
    increaseMonth: actions.increaseMonth
  }))(
    CssConnector(({ state }) => ({
      cssObject: state.TemplateCssObject
    }))(TemplateContainer)
  )
);
