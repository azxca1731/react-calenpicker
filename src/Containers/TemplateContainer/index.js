import TemplateContainer from "./TemplateContainer";

export default TemplateContainer;
