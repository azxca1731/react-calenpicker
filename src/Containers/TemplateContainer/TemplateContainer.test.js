import React from "react";
import { shallow } from "enzyme";
import TemplateContainer from "index";

describe("TemplateContainer", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<TemplateContainer />);
    expect(wrapper).toHaveLength(1);
  });
});
