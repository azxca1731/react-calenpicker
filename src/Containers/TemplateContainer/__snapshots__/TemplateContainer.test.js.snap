// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateContainer renders with snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
